src/core/operations/MultiTapCipherDecode.mjs

```diff
@@ -0,0 +1,65 @@
+/**
+ * @author Necron3574 [kaustubhbm3574@gmail.com]
+ * @copyright Crown Copyright 2021
+ * @license Apache-2.0
+ */
+
+import Utils from "../Utils.mjs";
+import Operation from "../Operation.mjs";
+import { getMultiTapMap } from "../lib/Ciphers.mjs";
+import OperationError from "../errors/OperationError.mjs";
+
+/**
+ * Multi Tap Cipher Decode operation
+ */
+class MultiTapCipherDecode extends Operation {
+
+    /**
+     * MultiTapCipherDecode constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Multi Tap Cipher Decode";
+        this.module = "Default";
+        this.description = "It is a simple substitution cipher which substitutes the alphabets to their corresponding taps on a mobile phone.<br><br>e.g. <code>adg</code> becomes <code>2-3-4</code>";
+        this.infoURL = "https://wikipedia.org/wiki/Multi-tap";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Delimiter",
+                type: "option",
+                value: ["Hyphen","Space", "Comma", "Semi-colon", "Colon", "Line feed"]
+            },
+            {
+              name : "Representation of Space",
+              type: "option",
+              value: ["Hash","Period","Percent","Ampersand"]
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        if(input.length == 0)
+          return "";
+        const delim = Utils.charRep(args[0] || "Hyphen");
+        const space_delim = Utils.charRep(args[1] || "Hash");
+        var dmap = getMultiTapMap(2,space_delim);
+        var enc_list = input.split(delim);
```

Comment:
I'm not sure that this will handle the decoding correctly - have you tested it on a range of inputs? For example, encoding the string `"ab d"` should produce `"2-22#3"` and then in this decoding that will be split into `"2"` and `"22#3"` as you are only splitting on the character delimiter, which means the last 2 letters won't be recognised. I think it needs to split on the space delimiter to get words, and then the character delimiter.

Reply:
Hey,
In the example you stated, `"ab d"` would produce `"2-22-#-3"` (Just tested this). This would again decode to `AB D`.
The encoding function will just treat a space as any other character and map it to its user chosen space delimiter(# in this case).
Similarly the decoding function maps the # back to a space.
I tested this on various inputs and it checked out.